Comp2024/src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,309 @@
+// ROBOTBUILDER TYPE: Subsystem.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.math.filter.MedianFilter;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.networktables.DoubleArraySubscriber;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.LLConsts;
+import frc.robot.Constants.VIConsts;
+
+/**
+ *
+ */
+public class Vision extends SubsystemBase
+{
+  // Objects
+  public MedianFilter           m_tyfilter      = new MedianFilter(5); // median filter y values to remove outliers (5 sample)
+  public MedianFilter           m_tvfilter      = new MedianFilter(5); // median filter v values to remove outliers (5 sample)
+
+  // Declare module variables
+  private double                m_distance1     = LLConsts.kDistance1;   // x position in inches for first reference point

```

Comment:
This group of variables is also about LL driving algorithm and should be deleted. (down to the constructor)